api-server/src/venues/gps.services.ts

```diff
@@ -1,64 +1,41 @@
-import {Injectable} from "@nestjs/common";
+import {HttpException, Injectable} from "@nestjs/common";
 import {Nullable} from "../utils/NullableOrUndefinable";
 import {GPSCoordinates} from "./dto/GetGPSCoordinatesRequest";
-import isNotNullOrUndefined from "../utils/is-not-null-or-undefined";
 import {isNullOrUndefined} from "../utils";
-import * as https from "https";
+import * as NodeGeocoder from "node-geocoder";
+import {MAP_BOX_API_KEY} from "../constants";
 
 @Injectable()
 export class GpsService {
-    async getCoordinatesFromGoogleMapLink(googleMapLink: string): Promise<Nullable<GPSCoordinates>> {
-        // see if the link has the coordinates embedded
-        const [hasCoordinatesInUrl, coordinateValues] = this.isUrlWithMatch(googleMapLink)
-
-        if (hasCoordinatesInUrl) {
-            return this.fromCoordinateValuesString(coordinateValues)
-        } else {
-            // some links are shortened, but will redirect to a full url with the coordinates embedded, use the full url
-            const resolvedRedirectUrl = await this.getRedirectUrl(googleMapLink)
-
-            const [hasCoordinatesInRedirectUrl, redirectCoordinateValues] = this.isUrlWithMatch(resolvedRedirectUrl)
-
-            return hasCoordinatesInRedirectUrl ? this.fromCoordinateValuesString(redirectCoordinateValues) : null
+    async getCoordinatesFromAddress({ street, city, state, zip }: AddressIdentifier): Promise<Nullable<GPSCoordinates>> {
+        if (isNullOrUndefined(MAP_BOX_API_KEY)) {
+            console.warn("attention developer, did you forget to set MAP_BOX_API_KEY")
+            throw new HttpException("No API Token Set", 403)
```

Comment:
Is this really a `403` / forbidden condition? It's not that the user isn't authorized, but rather that the server is misconfigured. Seems more like a `500` condition.

Reply:
yeah I could see that. I was just thinking 403 because auth can't be performed but either way works for me in this case

Reply:
I feel like this is not the first time I've landed on this post lol https://stackoverflow.com/questions/68559540/http-status-code-for-server-configuration-problem

Reply:
I suppose auth errors usually indicate the user can do something about it where as 5xx is usually something the dev/admin needs to looks at. I wish there was a more specific 500, but I think you're right 5xx is better. Going with 500 and a more full message

Reply:
662f15dc688545276eb5b314fbf06e164948e448